### ISSUE TITLE
Why does the v1.7.5 version report an error when it is running and the field definition is missing?

### ISSUE THREAD

**Issue by p9966** [gin-gonic/gin#2948]
# Description
***v1.7.5 Runtime error reporting missing fields*** 
![image](https://user-images.githubusercontent.com/32382395/142757747-ba867486-7eb9-4886-9f8b-8d69a64443f9.png)

# Expectations
Missing fields can be defined
> When I manually define the missing fields, the program can run normally

![image](https://user-images.githubusercontent.com/32382395/142757775-b1486de1-df53-448f-9509-0656ab3d643a.png)

# Environment
- gin version: v1.7.5

**Comment by aiyasin** [comment]
I have the same [problem](https://github.com/gin-gonic/gin/issues/2947) as you，and define the missing fields from your description, but the problem still exists.

**Comment by duyetpt** [comment]
I have the same problem as you

**Comment by DamiStudio** [comment]
I have got the same problem as you,it takes me several hours ,and finally i decided to give up trying fix it.

**Comment by Clivern** [comment]
I had the same issue 

```go
# github.com/gin-gonic/gin
Error: ../../../go/pkg/mod/github.com/gin-gonic/gin@v1.7.5/context.go:735:13: c.engine.AppEngine undefined (type *Engine has no field or method AppEngine)
Error: ../../../go/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:154:3: unknown field 'UseRawPath' in struct literal of type Engine
Error: ../../../go/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:156:3: unknown field 'UnescapePathValues' in struct literal of type Engine
Error: ../../../go/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:508:11: engine.UseRawPath undefined (type *Engine has no field or method UseRawPath)
Error: ../../../go/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:510:20: engine.UnescapePathValues undefined (type *Engine has no field or method UnescapePathValues)
Error: Process completed with exit code 2.
```

**Comment by vikinatora** [comment]
Just fixed this by following these steps(go 1.16.5):
1. require 1.7.2 in `go.mod`
`github.com/gin-gonic/gin v1.7.2`
2. Run `go mod tidy`
Now I have v1.7.5 installed with all the dependencies

**Comment by thinkerou** [comment]
@p9966 v1.7.5 don't release, only one branch.

**Comment by DamiStudio** [comment]
> @p9966 v1.7.5 don't release, it's an branch of git. I create the branch for release v1.7.5 but it have still error.

v1.7.2 works well.So you can delete the installed package of v 1.7.5,and reduce the version to v1.7.2.

**Comment by thinkerou** [comment]
v1.7.5 branch removed, thanks!

**Comment by Bisstocuz** [comment]
> v1.7.5 branch removed, thanks!

It may take some time to compare commits, because some commits with v1.7.5 label are based on some commits with v1.8.0 label.

**Comment by burik666** [comment]
@thinkerou, you should retract v1.7.5 in go.mod file.

https://golang.org/ref/mod#go-mod-file-retract

**Comment by thinkerou** [comment]
> @thinkerou, you should retract v1.7.5 in go.mod file.
> 
> https://golang.org/ref/mod#go-mod-file-retract

thanks! please see #2951

**Comment by gozeon** [comment]
<img width="1336" alt="image" src="https://user-images.githubusercontent.com/17582116/142856500-928bee4d-2648-4a88-b90e-7b78a8e57bb9.png">

**Comment by dineshnagarajandev** [comment]
Experienced similar issue with v1.7.5. 

Downgrading the version to github.com/gin-gonic/gin v1.7.2 resolves the issue.

**Comment by FahsaRwatit** [comment]
I see the same issue
go version go1.17.3 linux/amd64

`# github.com/gin-gonic/gin
/usr/local/gocode/pkg/mod/github.com/gin-gonic/gin@v1.7.5/context.go:735:13: c.engine.AppEngine undefined (type *Engine has no field or method AppEngine)
/usr/local/gocode/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:154:3: unknown field 'UseRawPath' in struct literal of type Engine
/usr/local/gocode/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:156:3: unknown field 'UnescapePathValues' in struct literal of type Engine
/usr/local/gocode/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:508:11: engine.UseRawPath undefined (type *Engine has no field or method UseRawPath)
/usr/local/gocode/pkg/mod/github.com/gin-gonic/gin@v1.7.5/gin.go:510:20: engine.UnescapePathValues undefined (type *Engine has no field or method Unescape`

**Comment by Zandut** [comment]
downgrade with v1.7.4 everything worked well

**Comment by thinkerou** [comment]
Released v1.7.6 using the code of v1.7.4 version to fix the error at first, please see https://github.com/gin-gonic/gin/releases/tag/v1.7.6, thanks!

**Comment by thinkerou** [comment]
v1.7.7 have released, thanks! https://github.com/gin-gonic/gin/releases
